packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -923,27 +941,23 @@ export class DeltaManager
         }
     }
 
-    private cancelDelayInfo(retryEndpoint: number) {
-        if (retryEndpoint === RetryFor.DeltaStorage) {
-            this.deltaStorageDelay = 0;
-        } else if (retryEndpoint === RetryFor.DeltaStream) {
-            this.deltaStreamDelay = 0;
-        }
+    public cancelDelayInfo(id: string) {
+        this.idToDelayMap.delete(id);
+        this.maxThrottlingDelay = Math.max(...this.idToDelayMap.values());
     }
 
-    private emitDelayInfo(retryEndpoint: number, delaySeconds: number, error: ICriticalContainerError) {
-        if (retryEndpoint === RetryFor.DeltaStorage) {
-            this.deltaStorageDelay = delaySeconds;
-        } else if (retryEndpoint === RetryFor.DeltaStream) {
-            this.deltaStreamDelay = delaySeconds;
-        }
-
-        const delayTime = Math.max(this.deltaStorageDelay, this.deltaStreamDelay);
-        if (delayTime > 0) {
+    public emitDelayInfo(
+        id: string,
+        delaySeconds: number,
+        error: ICriticalContainerError,
+    ) {
+        this.idToDelayMap.set(id, delaySeconds);
+        if (delaySeconds > 0 && delaySeconds > this.maxThrottlingDelay) {
```

Comment:
This is existing bug, but can you please open an issue to fix it?
The problem is that as time goes by this.maxThrottlingDelay becomes invalid. Instead we should be tracking point in time when it's going to end (i.e. not delay, but actual time).

Reply:
Opened the issue.